File: app/src/main/java/com/angogasapps/myfamily/ui/screens/main/holders/TextNewsViewHolder.java
package com.angogasapps.myfamily.ui.screens.main.holders;

import android.content.Context;
import android.view.View;

import androidx.annotation.NonNull;

import com.angogasapps.myfamily.databinding.TextNewsViewHolderBinding;
import com.angogasapps.myfamily.models.Family;
import com.angogasapps.myfamily.models.events.NewsObject;

public class TextNewsViewHolder extends BaseNewsViewHolder{
    private TextNewsViewHolderBinding binding;

    public TextNewsViewHolder(Context context, @NonNull View itemView) {
        super(context, itemView);
        binding = TextNewsViewHolderBinding.bind(itemView);
    }

    @Override
    public void update(NewsObject newsObject) {
        binding.text.setText(newsObject.getValue());
        binding.textName.setText(Family.getInstance().getNameByPhone(newsObject.getFromPhone()));
    }
}

File: app/src/main/java/com/angogasapps/myfamily/ui/screens/main/holders/TextNewsViewHolder.kt
package com.angogasapps.myfamily.ui.screens.main.holders

import android.content.Context
import android.view.View
import com.angogasapps.myfamily.databinding.TextNewsViewHolderBinding
import com.angogasapps.myfamily.ui.screens.main.holders.BaseNewsViewHolder
import com.angogasapps.myfamily.models.events.NewsObject
import com.angogasapps.myfamily.models.Family

class TextNewsViewHolder(context: Context, itemView: View) : BaseNewsViewHolder(
    context, itemView
) {
    private val binding: TextNewsViewHolderBinding = TextNewsViewHolderBinding.bind(itemView)
    override fun update(newsObject: NewsObject) {
        binding.text.text = newsObject.value
        binding.textName.text = Family.getInstance().getNameByPhone(newsObject.fromPhone)
    }

}